### PR TITLE
Feature/#31

### PR DESCRIPTION
**Pull request by jespernpedersen**
I've constructed a text filter to search on the dishes accordion menus. It emits a change to its parent component that triggers a function that sets the data.

### CODE REVIEW

**Review by akram0171 — APPROVED** [review]

File: src/components/AccordionMenu.vue [hunk 1]
```diff
@@ -60,6 +63,11 @@ export default {
                 filteredDishes[dish.course].dishes.push(dish);
             });
```
Comment (DiogoGeada) [inline src/components/AccordionMenu.vue]: Remove empty space

File: src/components/SearchBar.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        >
+        </v-text-field>
+        <button v-if="searchtext != ''" @click="clearFilter">
+            <figure>
+                <v-icon size="24px" >mdi-close</v-icon>
+            </figure>
+        </button>
+    </div>
+</template>
+<script>
```
Comment (DiogoGeada) [inline src/components/SearchBar.vue:18]: Add empty line between script and template

File: src/components/SearchBar.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export default {
+    name: 'SearchBar',
+    props: ['hasFiltedItems', 'dishes'],
```
Comment (DiogoGeada) [inline src/components/SearchBar.vue]: mispelt filtered

File: src/components/SearchBar.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    data () {
+        return {
+            searchtext: '',
+            isLoading: false,
```
Comment (DiogoGeada) [inline src/components/SearchBar.vue:25]: ~~This should come from the store. There is already a isLoadingDishes, isLoadingCourses and a method that returns true if either is still loading. You could use that instead. Create  computed property that returns that from the store~~
Nevermind, this is actually the correct way.

File: src/views/Home.vue [hunk 5]
```diff
@@ -1,12 +1,37 @@
...
+      @filter-dish="filterByText"
+      @clear-filter="clearFilter"
+    >
+    </SearchBar>
+    <h2>Menu</h2>
+    <div class="filtered-items" ref="filteredItems" v-show="filteredDishes.length > 0">
+      <AccordionMenu :dishes="filteredDishes" :courses="courses" :isLoading="isLoadingMenu" :activeByDefault="true"></AccordionMenu>
+    </div>
+    <div class="all-items" v-show="filteredDishes.length == 0">
+      <AccordionMenu :dishes="dishes" :courses="courses" :isLoading="isLoadingMenu" :activeByDefault="false"></AccordionMenu>
```
Comment (DiogoGeada) [inline src/views/Home.vue:21]: You're missing the title from the section (:title="Menu")

**Review by DiogoGeada — CHANGES_REQUESTED** [review]
You are missing an integration test for the searchbar. When you have time for it, feel free to contact me so that I can help you with it.

File: src/components/AccordionMenu.vue (hunk 1 above)
Comment (jespernpedersen) [reply src/components/AccordionMenu.vue]: Resolved

File: src/views/Home.vue (hunk 5 above)
Comment (jespernpedersen) [reply src/views/Home.vue:21]: We shouldn't put titles of sections into components, it prevents them from being reusable.

Comment (DiogoGeada) [reply src/views/Home.vue:21]: That is a very good point. I will mark as solved

**Review by DiogoGeada — APPROVED** [review]

**Pull request merged**